server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -290,6 +301,118 @@ long getAbsoluteStartMillis() {
         long buildTookInMillis() {
             return TimeUnit.NANOSECONDS.toMillis(relativeCurrentNanosProvider.getAsLong() - relativeStartNanos);
         }
+
+        public void setPhaseTookEnabled(boolean phaseTookEnabled) {
+            this.phaseTookEnabled = phaseTookEnabled;
+        }
+
+        public boolean isPhaseTookEnabled() {
+            return phaseTookEnabled;
+        }
+
+        SearchResponse.PhaseTook getPhaseTook() {
+            if (phaseTookEnabled) {
+                return new SearchResponse.PhaseTook(
+                    totalStats.dfsPreQueryTotal.count(),
+                    totalStats.canMatchTotal.count(),
+                    totalStats.queryTotal.count(),
+                    totalStats.fetchTotal.count(),
+                    totalStats.expandSearchTotal.count()
+                );
+            } else {
+                return SearchResponse.PhaseTook.NULL;
+            }
+        }
+
+        public SearchTimeProvider.RequestStatsHolder totalStats = new RequestStatsHolder();
+
+        public static final class RequestStatsHolder {
+            public CounterMetric dfsPreQueryTotal = new CounterMetric();
```

Comment:
These are latency metrics, so why use total in naming? And also use MeanMetrics considering this is related to latency.
Same for others.